sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_receiver_async.py

```diff
@@ -103,14 +133,48 @@ async def _outgoing_disposition(
         settled: Optional[bool],
         state: Optional[Union[Received, Accepted, Rejected, Released, Modified]],
         batchable: Optional[bool],
+        *,
+        message: Optional["_MessageDelivery"] = None,
+        on_disposition: Optional[Callable] = None,
     ):
         disposition_frame = DispositionFrame(
             role=self.role, first=first, last=last, settled=settled, state=state, batchable=batchable
         )
+
         if self.network_trace:
             _LOGGER.debug("-> %r", DispositionFrame(*disposition_frame), extra=self.network_trace_params)
+
+        # If trying to settle a message, keep track of the disposition
+        if message:
+            delivery = PendingDisposition(
+                message = message,
+                on_delivery_settled = on_disposition,
+            )
+            delivery.frame = disposition_frame
+            delivery.settled = settled
+            delivery.transfer_state = state
+            delivery.start = time.time()
+            delivery.sent = True
+
+            self._pending_receipts.append(delivery)
+
+
         await self._session._outgoing_disposition(disposition_frame) # pylint: disable=protected-access
 
+    async def _incoming_disposition(self, frame):
+        # If delivery_id is not settled, return
+        if not frame[3]:  # settled
```

Comment:
What does this mean? In what situation would it not be settled, and how would we then settle it?
Wondering if this could somehow result in a build-up in `self._pending_receipts` that never gets cleared and causes memory issues.